primitives/transaction-pool/src/pool.rs

```diff
@@ -210,8 +210,17 @@ pub trait TransactionPool: Send + Sync {
 	) -> PoolFuture<Box<TransactionStatusStreamFor<Self>>, Self::Error>;
 
 	// *** Block production / Networking
-	/// Get an iterator for ready transactions ordered by priority
-	fn ready(&self) -> Box<dyn Iterator<Item=Arc<Self::InPoolTransaction>>>;
+	/// Get an iterator for ready transactions ordered by priority.
+	///
+	/// Guarantees to return only when transaction pool got updated at `at` block.
+	/// Guarantees to return immediately when `None` is passed.
+	fn ready_at(&self, at: Option<NumberFor<Self::Block>>)
```

Comment:
Don't see the reason for `at` being an `Option` here.

If you don't want to have the `at` parameter, you can just call `ready`